landscaper/gardenlet/pkg/apis/imports/types.go

```diff
@@ -0,0 +1,90 @@
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package imports
+
+import (
+	"encoding/json"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	landscaperv1alpha1 "github.com/gardener/landscaper/apis/core/v1alpha1"
+)
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// Imports defines the landscaper import for the Gardenlet.
+type Imports struct {
+	metav1.TypeMeta
+	// SeedCluster contains the kubeconfig for the cluster
+	// - into which the Gardenlet is deployed by the landscaper
+	// - that is targeted as the Seed cluster by the Gardenlet via the default in-cluster mounted service account token
+	// Hence, the Gardenlet is always deployed into the Seed cluster itself.
+	// Deploying the Gardenlet outside of the Seed cluster is not supported (e.g., landscaper deploys Gardenlet into
+	// cluster A and the Gardenlet is configured via field .seedClientConnection.kubeconfig to target cluster B as Seed)
+	SeedCluster landscaperv1alpha1.Target
+	// GardenCluster is the landscaper target containing the kubeconfig for the
+	// Garden cluster (sometimes referred to as "virtual garden" - with Gardener API groups!)
+	GardenCluster landscaperv1alpha1.Target
+	// SeedBackup contains the credentials for an optional backup provider for the Seed cluster registered by the Gardenlet
+	// required when the Seed is configured for Backup (configured in the Gardenlet component configuration in the field seedConfig.spec.backup).
+	// before deploying the Gardenlet, the landscaper deploys a secret containing the specified credentials into the Garden cluster
+	SeedBackup *SeedBackup
+	// DeploymentConfiguration configures the Kubernetes deployment of the Gardenlet
+	DeploymentConfiguration *GardenletDeploymentConfiguration
+	// ComponentConfiguration specifies values for the Gardenlet component configuration
+	// This results in the configuration file loaded at runtime by the deployed Gardenlet
+	ComponentConfiguration runtime.Object
+}
+
+// SeedBackup contains configuration for an optional backup provider for the Seed cluster registered by the Gardenlet
+type SeedBackup struct {
+	// Provider is the provider name {aws,gcp,...}
+	Provider string
+	// Credentials contains provider specific credentials
+	// Please check the documentation of the respective extension provider for the concrete format
+	Credentials json.RawMessage
+}
+
+// GardenletDeploymentConfiguration specifies Gardenlet deployment parameters, such as the number of replicas,
+// the required resources, etc.
+type GardenletDeploymentConfiguration struct {
```

Comment:
It is indeed a copy of _seedmanagement.GardenletDeployment_ minus the image (comes from the component descriptor here) and the two image vectors. 

I think the image vectors for the managed seed could only be dropped because it gets the image vectors from its own Gardenlet deployment right? But this is not possible here.

So we have three different fields here - I was not sure if that justifies a common field. But I am open to it. 
Maybe something like _gardenletbaseconfig.GardenletDeploymentConfiguration_ ?

Reply:
I am not sure about ImageVectorOverwrite and ComponentImageVectorOverwrites for the managed seed actually - is that useful?. Do we need the ability to specify a different image vector per managed seed?

Reply:
@rfranzke @timebertt do you guys have an opinion here? Then I can change the API accordingly (or not)

Reply:
> Is that useful for the managed seed?. Do we need the ability to specify a different image vector per managed seed?
I would say yes, at least I would feel better if we had it there. It was removed because it was argued it would be rarely used (most of the time getting it from parent is what we want), and we were not sure about the types and didn't want to spend additional time to figure them out.
But seeing that we are in a rather similar situation here, I would suggest to clarify this, add them to the "base deployment configuration" (whatever that type is) and reuse this type both here and for managed seeds.

Reply:
Or we could add the image vector overwrites to `seedmanagement.GardenletDeployment`, and instead of having a base type, we could just reuse this type. Regarding the image, you could add a validation check that it's empty in your case.

Reply:
ok that is fine for me. I will add it to the seedmanagement.GardenletDeployment and reuse it

Reply:
> I could contribute a separate PR adding these 2 fields to seedmanagement.GardenletDeployment, after we clarify what type they should be (@timebertt had objections to using just `string`), making it easier for you to reuse it.

Yes, this sounds like a good plan. This use case would now justify adding `imageVectorOverwrite` to `seedmanagement.GardenletDeployment` IMO. And then the landscaper component can reuse the seedmanagement type.
I would still argue, that we should have proper API types instead of using plain string for those fields, as the values' structure is well-known and they should be represented in well-typed API types.

Reply:
I temporarily changed the seedmanagement.GardenletDeployment to include the image vector as a string.
Will rebase on top of the PR that @stoyanr is preparing to introduce strong types for the image vector.

Reply:
The PR is already open (#3656), you can use it once it's merged.

Reply:
We have discusses to not rebase on top of this PR, but rather reused the seedmanagement GardenletDeployment field, but have dedicated image vector fields in the import configuration of the Gardenlet landscaper-